keel-titus-plugin/src/test/kotlin/com/netflix/spinnaker/keel/titus/verification/ContainerTestVerificationEvaluatorTests.kt

```diff
@@ -0,0 +1,164 @@
+    artifactReference = "fnord",
+    version = "1.1"
+  )
+  private val verification = ContainerTestVerification(
+    repository = "illuminati/fnord",
+    location = Location(
+      account = "titustestvpc",
+      region = "ap-south-1"
+    )
+  )
+
+  @Test
+  fun `starting verification launches a container job via orca`() {
+    val taskId = ULID().nextULID()
+
+    every {
+      taskLauncher.submitJob(
+        user = any(),
+        application = any(),
+        notifications = any(),
+        subject = any(),
+        description = any(),
+        correlationId = any(),
+        stages = any()
+      )
+    } answers { Task(id = taskId, name = arg(3)) }
+
+    expectCatching {subject.start(context, verification)}
+      .isSuccess()
+      .get("taskId") isEqualTo taskId
+
+    verify {
+      taskLauncher.submitJob(
+        user = any(),
+        application = any(),
+        notifications = any(),
+        subject = any(),
+        description = any(),
+        correlationId = any(),
+        stages = match {
+          it.first()["type"] == RUN_JOB_TYPE
+        }
+      )
+    }
+  }
+
+  @ParameterizedTest(name = "verification is considered still running if orca status is {0}")
+  @EnumSource(
+    OrcaExecutionStatus::class,
+    names = ["NOT_STARTED", "RUNNING", "PAUSED", "SUSPENDED", "BUFFERED"]
+  )
+  fun `verification is considered still running if orca task is running`(taskStatus: OrcaExecutionStatus) {
+    val taskId = ULID().nextULID()
+    val previousState = runningState(taskId)
+
+    every {
+      orca.getOrchestrationExecution(taskId, any())
+    } returns ExecutionDetailResponse(
+      id = taskId,
+      name = "whatever",
+      application = "fnord",
+      buildTime = previousState.startedAt,
+      startTime = previousState.startedAt,
+      endTime = null,
+      status = taskStatus
+    )
+
+    expectThat(subject.evaluate(context, verification, previousState.metadata)) isEqualTo RUNNING
+  }
+
+  @ParameterizedTest(name = "verification is considered successful if orca status is {0}")
+  @EnumSource(
+    OrcaExecutionStatus::class,
+    names = ["SUCCEEDED"]
+  )
+  fun `verification is considered successful if orca task succeeded`(taskStatus: OrcaExecutionStatus) {
+    val taskId = ULID().nextULID()
+    val previousState = runningState(taskId)
+
+    every {
+      orca.getOrchestrationExecution(taskId, any())
+    } returns ExecutionDetailResponse(
+      id = taskId,
+      name = "whatever",
+      application = "fnord",
+      buildTime = previousState.startedAt,
+      startTime = previousState.startedAt,
+      endTime = null,
+      status = taskStatus
+    )
+
+    expectThat(subject.evaluate(context, verification, previousState.metadata)) isEqualTo PASSED
+  }
+
+  @ParameterizedTest(name = "verification is considered failed if orca status is {0}")
+  @EnumSource(
+    OrcaExecutionStatus::class,
+    names = ["TERMINAL", "FAILED_CONTINUE", "STOPPED", "CANCELED"]
```

Comment:
You sure about the spelling of `CANCELED` in orca? I didn't look, but caught my eye... 🙂

Reply:
Never mind, you got it right.

Reply:
Yeah, I copied these from the enum. It's very annoying you can't use an enum in annotations.